File: buerguer-pig/AMarotoChivite-BurguerPig/AMarotoChivite-BurguerPig-Java/src/main/java/Ficheros/BurguerPig/mappers/BurguerMapper.java
package Ficheros.BurguerPig.mappers;

import Ficheros.BurguerPig.mappers.base.IDtoMapper;
import Ficheros.BurguerPig.mappers.base.ListMapper;
import Ficheros.BurguerPig.models.Burguer;
import Ficheros.BurguerPig.models.Ingredient;
import Ficheros.BurguerPig.models.dto.BurguerDTO;
import Ficheros.BurguerPig.models.dto.IngredienteDTO;

import java.util.List;
import java.util.UUID;
import java.util.stream.Collectors;

public class BurguerMapper implements IDtoMapper<BurguerDTO, Burguer> {

    @Override
    public BurguerDTO toDto(Burguer model) {
        return new BurguerDTO(
                model.getName(),
                //model.getIngredients().stream().map(it -> new IngredienteDTO(it.getName(), String.valueOf(it.getPrice()), String.valueOf(it.getID()))).collect(Collectors.toList()),
                new IngredienteListMapper(new IngredienteMapper()).toDto(model.getIngredients()).list,
                model.getUuid().toString(),
                String.valueOf(model.getPrice())
        );
    }

    @Override
    public Burguer toModel(BurguerDTO dto) {
        return new Burguer(
                dto.getName(),
                //dto.getIngredients().stream().map((it -> new Ingredient(it.getName(),Double.parseDouble(it.getPrice()), Integer.parseInt(it.getId())))).collect(Collectors.toList()),
                new IngredienteListMapper(new IngredienteMapper()).toModelList(dto.getIngredients()),
                UUID.fromString(dto.getUuid()),
                Double.parseDouble(dto.getPrice())
        );
    }
}

File: buerguer-pig/AMarotoChivite-BurguerPig/AMarotoChivite-BurguerPig-Kotlin/src/main/kotlin/Ficheros/BurguerPig/mappers/BurguerMapper.kt
package mappers

import Ficheros.BurguerPig.mappers.base.IDtoMapper
import Ficheros.BurguerPig.mappers.base.ListMapper
import Ficheros.BurguerPig.models.Burguer
import Ficheros.BurguerPig.models.dto.BurguerDTO
import Ficheros.BurguerPig.models.dto.BurguerListDto
import java.util.*

class BurguerMapper : IDtoMapper<BurguerDTO, Burguer> {

    override fun toDto(model: Burguer): BurguerDTO {
        return BurguerDTO(
            model.name,
            IngredientListMapper().toDto(model.ingredients).list,
            model.getUUID().toString(),
            model.getPrice().toString()
        )
    }

    override fun toModel(dto: BurguerDTO): Burguer {
        return Burguer(
            dto.name,
            IngredientListMapper().toModelList(dto.ingredients),
            UUID.fromString(dto.uuid),
            dto.price.toDouble()
        )
    }
}

class BurguerListMapper : ListMapper<BurguerDTO, Burguer>(BurguerMapper()) {

    fun toDtoList(modelList: List<Burguer>): BurguerListDto {
        val dtoList = modelList.map { BurguerMapper().toDto(it) }
        return BurguerListDto(dtoList)
    }

    fun toModelList(dtoList: BurguerListDto): List<Burguer> {
        return dtoList.burguers.map { BurguerMapper().toModel(it) }
    }
}


